### DISCUSSION TITLE
Inclure une nouvelle librairie (Ox_Lib)

### DISCUSSION THREAD

**Discussion by spacevx** [SOZ-Faut-etre-Sub/SOZ-FiveM-Server#62] [Ideas]
Ox_lib est une libraire faite par le groupe de développeurs Ox (ceux ayant fait la ressource oxmysql)

Elle permet de faire beaucoup de chose, mais son point fort est l'optimisation elle utilise lua 5.4 et l'argument fx_v2-oal (vous pouvez voir <a href="https://forum.cfx.re/t/lua-5-4-performance-issues-or-just-potato/4755067">ici</a> les différents benchmarks et comment marche le fx_v2-oal), elle permet de faire une optimisation non-négligeable au lieu d'utiliser la native `GetDistanceBetweenCoords`
ou de faire le simple calcul mathématique 
```lua
local distance = #(coords1 - coords2);
```
Cette libraire contient un système de <a href="https://overextended.github.io/docs/ox_lib/Points/Client">points</a> qui fait en interne les calculs de distance de façon très optimisée.

La libraire possède aussi un système de <a href="https://overextended.github.io/docs/ox_lib/Zones/Client">Zone</a> comme PolyZone mais en + optimisé, je suggère de rajouter cette libraire et de l'adapter aux différentes ressources du serveur petit à petit, pour pouvoir au final supprimé PolyZone qui est une ressource assez ancienne maintenant.

Elle possède un système de cache qui permet de garder en cache le ped `PlayerPedId()` et d'autre chose (voir <a href="https://overextended.github.io/docs/ox_lib/Cache/Client">ici</a>)

Pleins d'autres fonctionnalités sont disponibles, vous pouvez consulter la <a href="https://overextended.github.io/docs/category/library">documentation</a> et voici le <a href="https://github.com/overextended/ox_lib">repo</a>

**Comment by joelwurtz** [comment]
Merci de la proposition mais ce n'est clairement pas la volonté d'inclure plus de lib en lua, nous migrons au fur et a mesure nos modules en lua vers le module soz-core écrit en typescript.

A savoir que la plupart des propositions que tu as cité sont déjà présentes dans le core (on a plus ou moins refait notre propre système de zone + de calcul de coordonnées et qui assez optimisé)

En ce qui concerne le PlayerPedId() je suis assez contre le principe de mettre du cache par dessus des natives (sauf cas exceptionnel), car au final on n'économise que quelques chemins de fonction et c'est une optimisation très faible (de l'ordre de quelque nano secondes) pour une complexité accru (invalider du cache = prise de tête)
